File: js-mocha/test/texttest_fixture.js
const { Shop, Item } = require("../src/gilded_rose");

exports.compose = function (days = 30) {
  const lines = [];

  const items = [
    new Item("+5 Dexterity Vest", 10, 20),
    new Item("Aged Brie", 2, 0),
    new Item("Elixir of the Mongoose", 5, 7),
    new Item("Sulfuras, Hand of Ragnaros", 0, 80),
    new Item("Sulfuras, Hand of Ragnaros", -1, 80),
    new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
    new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
    new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),

    // This Conjured item does not work properly yet
    new Item("Conjured Mana Cake", 3, 13),
  ];

  const gildedRose = new Shop(items);

  lines.push("OMGHAI!");
  for (let day = 0; day < days; day++) {
    lines.push(`\n-------- day ${day} --------`);
    lines.push("name, sellIn, quality");
    items.forEach((item) =>
      lines.push(`${item.name}, ${item.sellIn}, ${item.quality}`)
    );
    gildedRose.updateQuality();
  }

  return lines.join("\n");
};

File: csharpcore/GildedRoseTests/TextTestFixture.cs
using System;
using System.Collections.Generic;
using GildedRoseKata;

namespace GildedRoseTests;

public static class TextTestFixture
{
    public static void Main(string[] args)
    {
        Console.WriteLine("OMGHAI!");

        var items = new List<Item>{
            new ItemBuilder("+5 Dexterity Vest", 10,20).Build(),
            new ItemBuilder("Aged Brie", 2, 0).Build(),
            new ItemBuilder("Elixir of the Mongoose", 5, 7).Build(),
            new LegendaryItemBuilder("Sulfuras, Hand of Ragnaros", 0).Build(),
            new LegendaryItemBuilder("Sulfuras, Hand of Ragnaros", -1).Build(),
            new ItemBuilder("Backstage passes to a TAFKAL80ETC concert",
                15,
                20
                ).Build(),
            new ItemBuilder("Backstage passes to a TAFKAL80ETC concert",
                10,
                49
                ).Build(),
            new ItemBuilder("Backstage passes to a TAFKAL80ETC concert",
                5,
                49
                ).Build(),
            // this conjured item does not work properly yet
            new ItemBuilder("Conjured Mana Cake", 3, 6).Build()
        };

        var app = new GildedRose(items);

        int days = 2;
        if (args.Length > 0)
        {
            days = int.Parse(args[0]) + 1;
        }

        for (var i = 0; i < days; i++)
        {
            Console.WriteLine("-------- day " + i + " --------");
            Console.WriteLine("name, sellIn, quality");
            for (var j = 0; j < items.Count; j++)
            {
                Console.WriteLine(items[j].Name + ", " + items[j].SellIn + ", " + items[j].Quality);
            }
            Console.WriteLine("");
            app.UpdateQuality();
        }
    }
}